src/super_gradients/modules/detection_modules.py

```diff
@@ -20,10 +27,361 @@ def __init__(self, in_channels: Union[List[int], int], **kwargs):
+        neck2: Union[str, HpmStruct, DictConfig],
+        neck3: Union[str, HpmStruct, DictConfig],
+        neck4: Union[str, HpmStruct, DictConfig],
+    ):
+        super().__init__(in_channels)
+        c3_out_channels, c4_out_channels, c5_out_channels = in_channels
+
+        factory = det_factory.DetectionModulesFactory()
+        self.neck1 = factory.get(factory.insert_module_param(neck1, "in_channels", [c5_out_channels, c4_out_channels]))
+        self.neck2 = factory.get(factory.insert_module_param(neck2, "in_channels", [self.neck1.out_channels[1], c3_out_channels]))
+        self.neck3 = factory.get(factory.insert_module_param(neck3, "in_channels", [self.neck2.out_channels[1], self.neck2.out_channels[0]]))
+        self.neck4 = factory.get(factory.insert_module_param(neck4, "in_channels", [self.neck3.out_channels, self.neck1.out_channels[0]]))
+
+        self._out_channels = [
+            self.neck2.out_channels[1],
+            self.neck3.out_channels,
+            self.neck4.out_channels,
+        ]
+
+    @property
+    def out_channels(self):
+        return self._out_channels
+
+    def forward(self, inputs):
+        c3, c4, c5 = inputs
+
+        x_n1_inter, x = self.neck1([c5, c4])
+        x_n2_inter, p3 = self.neck2([x, c3])
+        p4 = self.neck3([p3, x_n2_inter])
+        p5 = self.neck4([p4, x_n1_inter])
+
+        return p3, p4, p5
+
+
+class NHeads(BaseDetectionModule):
+    """
+    Apply N heads in parallel and combine predictions into the shape expected by SG detection losses
+    """
+
+    def __init__(self, in_channels: List[int], num_classes: int, heads_list: Union[str, HpmStruct, DictConfig]):
+        super().__init__(in_channels)
+        factory = det_factory.DetectionModulesFactory()
+        heads_list = self._pass_num_classes(heads_list, factory, num_classes)
+
+        self.num_heads = len(heads_list)
+        for i in range(self.num_heads):
+            new_head = factory.get(factory.insert_module_param(heads_list[i], "in_channels", in_channels[i]))
+            setattr(self, f"head{i + 1}", new_head)
+
+    @staticmethod
+    def _pass_num_classes(heads_list, factory, num_classes):
+        for i in range(len(heads_list)):
+            heads_list[i] = factory.insert_module_param(heads_list[i], "num_classes", num_classes)
+        return heads_list
+
+    @property
+    def out_channels(self):
+        return None
+
+    def forward(self, inputs):
+        outputs = []
+        for i in range(self.num_heads):
+            outputs.append(getattr(self, f"head{i + 1}")(inputs[i]))
+
+        return self.combine_preds(outputs)
+
+    def combine_preds(self, preds):
+        outputs = []
+        outputs_logits = []
+        for output, output_logits in preds:
+            outputs.append(output)
+            outputs_logits.append(output_logits)
+
+        return outputs if self.training else (torch.cat(outputs, 1), outputs_logits)
+
+
+class MobileNetV1Backbone(BaseDetectionModule):
```

Comment:
MobileNetV1Backbone and MobileNetV2Backbone are almost identical, can we perhaps introduce a parent class ?

Reply:
Agree, done e6172c5